### DISCUSSION TITLE
[Question] What are the differences in the method of spawning assets in multiple environments between the Standalone code and RL code in IsaacLab?

### DISCUSSION THREAD

**Discussion by choiwanuk** [isaac-sim/IsaacLab#1601] [Q&A]
Hello,

I am currently attempting to use an asset combining deformable and articulation components in reinforcement learning through IsaacLab. While I came across a post in the NVIDIA Developer Community stating that this case is currently unsupported, I believe exploring this scenario could yield interesting results. While searching for ways to enable this in IsaacLab, I observed different outcomes when spawning the complex asset in the Standalone code versus the RL code.

The complex asset used for testing consists of an articulation made up of two rigid rod links connected by a revolute joint, with a deformable component attached.

 In the RL code (train.py), when I created multiple num_envs with this asset, the following error occurred, causing the simulation to function properly in only one environment while failing in the others. (Related video attachment below)

`
Replication of this type is not supported: ~,prim path: ~/deformable \n 
Replication of this type is not supported: ~,prim path: ~/attachment  
`

https://github.com/user-attachments/assets/e5e2aa0b-1204-4847-a3e5-cebc82ac47f4

In contrast, when spawning an asset with the same structure in multiple environments using the Standalone code, the simulation ran without any issues under the same conditions. (Related video attachment below)

https://github.com/user-attachments/assets/585d86a5-7ff5-4fbc-8347-d6ce69de8efe

Below are my questions regarding this phenomenon:

1. I understand that both codes are based on the same API. Could the difference in outcomes be related to the creation of state tensors in RL for use in observations and rewards?

2. Is it challenging to modify the RL code to enable the simulation to work? If not, which parts should I adjust?

3. Would it be feasible to integrate a reinforcement learning framework like rl_games into the Standalone code?

Thank you for taking the time to read my inquiry.

**Comment by RandomOakForest** [comment]
Thank you for posting this. We have had some delay with the holidays. We should be able to reply to this soon.

**Comment by StrainFlow** [comment]
Are you using a managed or direct environment? Can you share some of your code? 

For something like this, you might also want to join us on discord. There's a group that are all trying to do interesting things with Isaac Lab and you might find like-minded people who want to work with you on more experimental stuff like this: 

https://discord.gg/bNuRw4nJXW
